pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/relocation/SegmentRelocator.java

```diff
@@ -196,6 +204,14 @@ private void rebalanceTable(String tableNameWithType) {
     rebalanceConfig.setMinimizeDataMovement(_minimizeDataMovement);
 
     try {
+      if (_tablesUndergoingRebalance != null) {
```

Comment:
We should move this outside of the `try-catch` block. When early terminated, we shouldn't call `remove()`

Reply:
yup i just realized that and updated